framework/console/controllers/AssetController.php

```diff
@@ -326,6 +327,11 @@ protected function loadTargets($targets, $bundles)
     protected function buildTarget($target, $type, $bundles)
     {
         $tempFile = $target->basePath . '/' . strtr($target->$type, ['{hash}' => 'temp']);
+        if (!is_file($tempFile)) {
+            FileHelper::createDirectory(dirname($tempFile), 0755, true);
+            touch($tempFile);
```

Comment:
If  even after fix with creating unexisting path this problem appears, it could happen only if there are no files to be compressed like in case they all are external links.
In this case the check for this should be added before `file_md5` instead of just creating empty file.